src/proto/h1/role.rs

```diff
@@ -1441,9 +1442,17 @@ fn write_headers_title_case(headers: &HeaderMap, dst: &mut Vec<u8>) {
     }
 }
 
-fn write_headers(headers: &HeaderMap, dst: &mut Vec<u8>) {
+fn write_headers<'a>(headers: &HeaderMap, dst: &mut Vec<u8>, special_headers: Option<&'a HashMap<&'static str, &'static [u8]>>) {
```

Comment:
I don't think we actually need the `'a`, removing it should be fine.
```suggestion
fn write_headers(headers: &HeaderMap, dst: &mut Vec<u8>, special_headers: Option<&HashMap<&'static str, &'static [u8]>>) {
```